src/main/java/mk/p/singleton/v4/MySingleton.java

```diff
@@ -0,0 +1,15 @@
+package mk.p.singleton.v4;
+
+public class MySingleton {
+    private MySingleton() {
+        System.out.println("Singleton created");
+    }
+
+    private static class SingletonHolder {
+        public static final MySingleton HOLDER_INSTANCE = new MySingleton();
+    }
```

Comment:
Why the inner class definition is between methods of the parent class?

Reply:
Why not?

Reply:
Review java code convention doc